### ISSUE TITLE
Wrong report of TS2525

### ISSUE THREAD

**Issue by n9** [Microsoft/TypeScript#26574]
**TypeScript Version:**  3.0.1 and also current `@next` 3.1.0-dev.20180821

**Search Terms:** TS2525

**Code**

```ts
interface Foo {
    b?: string
}

function bar1(arg?: Foo): void {
    const { b } = arg || {}
}

function bar2(arg?: Foo): void {
    const da = arg || {}
    const { b } = da
}
```

**Expected behavior:**
No error.

**Actual behavior:**
`bar1` shows error: *TS2525: Initializer provides no value for this binding element and the binding element has no default value.*

**Playground Link:** https://www.typescriptlang.org/play/#src=interface%20Foo%20%7B%0D%0A%20%20%20%20b%3F%3A%20string%0D%0A%7D%0D%0A%0D%0Afunction%20bar1(arg%3F%3A%20Foo)%3A%20void%20%7B%0D%0A%20%20%20%20const%20%7B%20b%20%7D%20%3D%20arg%20%7C%7C%20%7B%7D%0D%0A%7D%0D%0A%0D%0Afunction%20bar2(arg%3F%3A%20Foo)%3A%20void%20%7B%0D%0A%20%20%20%20const%20da%20%3D%20arg%20%7C%7C%20%7B%7D%0D%0A%20%20%20%20const%20%7B%20b%20%7D%20%3D%20da%0D%0A%7D

**Related Issues:** no (only https://github.com/Microsoft/TypeScript/issues/21634)

**Comment by AlCalzone** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/26235?

**Comment by n9** [comment]
@AlCalzone Yes, seems to.
